src/main.js

```diff
@@ -0,0 +1,271 @@
+import { query, queryDeletedCases } from "./lib/api";
+import { escapeHTML, formatAddress, formatStatus } from "./lib/format";
+
+const CLASSES = {
+  container: "sirius-search",
+  list: "sirius-search__list",
+  item: "sirius-search__item",
+  itemSummary: "sirius-search__item--summary",
+  link: "sirius-search__link",
+};
+
+function SearchResults($input, $form) {
+  if ($input.searchResults) {
+    return $input.searchResults.search();
+  }
+
+  $input.searchResults = this;
+
+  this.$input = $input;
+  this.$form = $form;
+  this.config = this.getConfig();
+
+  this.$container = document.createElement("div");
+  this.$container.classList.add("govuk-body", CLASSES.container);
+
+  let inputBoundingBox = this.$input.getBoundingClientRect();
+  if (this.config.attach) {
+    inputBoundingBox = document
+      .querySelector(this.config.attach)
+      .getBoundingClientRect();
+  }
+
+  this.$container.style.top = `${inputBoundingBox.bottom}px`;
+  this.$container.style.left = `${inputBoundingBox.left}px`;
+  this.$container.style.width = `${inputBoundingBox.width}px`;
+
+  if (this.config.position === "fixed") {
+    this.$container.style.position = "fixed";
+  }
+
+  this.$form.appendChild(this.$container);
+
+  this.$input.addEventListener("focus", () => {
+    this.open();
+  });
+
+  this.$input.addEventListener("keydown", (e) => {
+    if (e.key === "Escape") {
+      this.close();
+    }
+  });
+
+  document.addEventListener("click", (e) => {
+    if (!$form.contains(e.target)) {
```

Comment:
what is the difference for adding the event listener to `document` and filtering vs. adding it to the element itself?

Reply:
It's specifically checking for clicks that _aren't_ in `$form` so that it can close the popup when you click something else.

Reply:
ugh yes, I will blame my eyesight